diff -up netkit-rusers-0.17/rpc.rstatd/rstatd.c.include netkit-rusers-0.17/rpc.rstatd/rstatd.c
--- netkit-rusers-0.17/rpc.rstatd/rstatd.c.include	1999-02-10 20:27:53.000000000 +0100
+++ netkit-rusers-0.17/rpc.rstatd/rstatd.c	2012-02-06 13:36:30.395135558 +0100
@@ -40,6 +40,8 @@ static char rcsid[] = "$Id: rstatd.c,v 1
 #include <signal.h>
 #include <syslog.h>
 #include <rpcsvc/rstat.h>
+#include <sys/socket.h>
+#include <unistd.h>
 
 extern void rstat_service();
 
diff -up netkit-rusers-0.17/rpc.rusersd/rusers_proc.c.include netkit-rusers-0.17/rpc.rusersd/rusers_proc.c
--- netkit-rusers-0.17/rpc.rusersd/rusers_proc.c.include	2012-02-06 13:36:23.922135996 +0100
+++ netkit-rusers-0.17/rpc.rusersd/rusers_proc.c	2012-02-06 13:36:30.314135559 +0100
@@ -31,7 +31,7 @@ char rp_rcsid[] =
 
 #include <signal.h>
 #include <sys/types.h>
-#include <sys/time.h>
+#include <time.h>
 #include <utmp.h>
 #include <stdio.h>
 #include <syslog.h>
diff -up netkit-rusers-0.17/rup/err.c.include netkit-rusers-0.17/rup/err.c
--- netkit-rusers-0.17/rup/err.c.include	1997-04-06 00:13:31.000000000 +0200
+++ netkit-rusers-0.17/rup/err.c	2012-02-06 13:36:30.314135559 +0100
@@ -6,6 +6,7 @@ char err_rcsid[] =
   "$Id: err.c,v 1.1 1997/04/05 22:13:31 dholland Exp $";
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <stdarg.h>
 #include <err.h>
 
diff -up netkit-rusers-0.17/rusers/rusers.c.include netkit-rusers-0.17/rusers/rusers.c
--- netkit-rusers-0.17/rusers/rusers.c.include	1999-12-12 20:32:05.000000000 +0100
+++ netkit-rusers-0.17/rusers/rusers.c	2012-02-06 13:36:30.355135558 +0100
@@ -44,6 +44,7 @@ char rusers_rcsid[] =
 #include <utmp.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <time.h>
 
 #include "../version.h"
 
